mlos_bench/mlos_bench/environments/composite_env.py

```diff
@@ -52,6 +52,13 @@ def __init__(self,
 
         self._children: List[Environment] = []
 
+        # To support trees of composite environments (e.g. for multiple VM experiments),
+        # each CompositeEnv gets a copy of the original global config and adjusts it with
+        # the `const_args` specific to it.
+        global_config = (global_config or {}).copy()
+        for (key, val) in self._const_args.items():
+            global_config.setdefault(key, val)
```

Comment:
Tests for this change?

Reply:
e.g. a sample CompositeEnv tree config and then make changes to each side and not the other and verify that the parent and sibling was not altered.

Reply:
you're right. will add now